game-engine/src/commonMain/kotlin/za/co/entelect/challenge/game/engine/renderer/WormsRendererConsole.kt

```diff
@@ -15,10 +15,10 @@ class WormsRendererConsole(private val config: GameConfig) : WormsRenderer {
         return """Player ${wormsPlayer.id}, enter a command (move x y)/(dig x y)/(shoot $directionsString)/(nothing)"""
     }
 
-    override fun render(wormsMap: WormsMap, player: WormsPlayer): String {
-        val wormPosition = player.currentWorm.position
+    override fun render(wormsMap: WormsMap, player: WormsPlayer?): String {
+        val wormPosition = player!!.currentWorm.position
```

Comment:
If player is never null here, whats the point of making the param nullable? I think it might make more sense to make it non-null and ensure it isn't null wherever you are calling it from...

Reply:
if `player: WormsPlayer?` is changed, it no longer overrides the same function 😔

Reply:
Does global state renderer have to use the exact same interface? Can it not be a completely different interface with nullable helper methods where it makes sense?

Otherwise, just move the null check to the beginning of the method and throw an exception or something if it is null. That will clarify a lot, I think.

Reply:
The null check is far easier, thanks! 😄 
The extra interface muddies the game-runner trying to do this. If this turns out to be really useful, we should look at how to implement it into the permanent projects